core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/KeyStoreManager.java

```diff
@@ -487,19 +373,12 @@ public String getPassword(org.wso2.carbon.registry.core.Resource resource) throw
      */
     public String getKeyStorePassword(String keyStoreName) throws Exception {
 
-        String path = RegistryResources.SecurityManagement.KEY_STORES + "/" + keyStoreName;
-        if (registry.resourceExists(path)) {
-            Resource resource = registry.get(path);
+        String encryptedPassword = keyStoreRegistryPersistenceManager.getEncryptedKeyStorePassword(keyStoreName);
+        if (encryptedPassword != null) {
```

Comment:
Stringutils.isnotblank

Reply:
Shall we do the null check inside the method and return the decrypted password?

Reply:
Fixed in the latest changes